pandas/core/util/hashing.py

```diff
@@ -276,3 +302,23 @@ def hash_array(vals, encoding='utf8', hash_key=None, categorize=True):
     vals *= np.uint64(0x94d049bb133111eb)
     vals ^= vals >> 31
     return vals
+
+
+def _hash_scalar(val, encoding='utf8', hash_key=None):
+    """
+    Hash scalar value
+
+    Returns
+    -------
+    1d uint64 numpy array of hash value, of length 1
+    """
+
+    if isnull(val):
+        # this is to be consistent with the _hash_categorical implementation
+        return np.array([np.iinfo(np.uint64).max], dtype='u8')
+
```

Comment:
if you need to handle datetime w/tz directly (IOW, we basically ignore it), then I would.

```
if getattr(val, 'tzinfo', None) is not None:
    val = val.tz_localize(None)
```

I suppose an option to ignore tz is fine for ``infer_dtype_from_scalar``, but if you add it I would rename, document and test.